### PR TITLE
build: use node 14 or higher

### PR DESCRIPTION
**Pull request by Kikobeats**
The main point of this PR is

- Avoid to maintained out of LTS node builds
- Remove monkey patch code (`String.prototype.startsWith`) since is not more necessary
- Reduce code using `util.promisify`

### CODE REVIEW

**Review by huan — APPROVED** [review]
LGTM

**Comment by Kikobeats** [comment]
not sure why windows builds are failing, probably is related to a thing not related with this PR  🤔

**Comment by simonepri** [comment]
Thanks for contributing!

At the moment this DLGTM.
I think is appropriate to bump the package to a stable before dropping support for node 0.10 (altough we don't have the CI testing it, it is still supported).

Before doing this though, I need to improve the CI and find a way to test on more OS than we currently do (maybe using a mixure of vagrant and docker, suggestions welcome).

**Comment by Kikobeats** [comment]
@simonepri I agree, right now CI build is a problem.

What if we delay this to be shipped under `1.0.0` version? 

Then both things could be done: If you need to use Node 10 or lower, use 0.x, otherwise 1.x is for you.

**Comment by simonepri** [comment]
Yes that's exactly the plan!

The PR per se looks good, it is just that some other work needs to be done before this can be merged.

**Comment by huan** [comment]
@simonepri I have merged the #14 because that improvement will not breaking anything.

However, the Node version 4/6 are removed from the CI testing by that PR.

FYI.

File: .github/workflows/test-macos.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        node: [12, 10, 8, 6, 4]
+        node: [15, 14, 12, 10]
```
Comment (jimmywarting) [inline .github/workflows/test-macos.yml]:
```diff
-        node: [15, 14, 12, 10]
+        node: [14, 16, 18]
```

File: .github/workflows/test-ubuntu.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        node: [12, 10, 8, 6, 4]
+        node: [15, 14, 12, 10]
```
Comment (jimmywarting) [inline .github/workflows/test-ubuntu.yml]:
```diff
-        node: [15, 14, 12, 10]
+        node: [14, 16, 18]
```

File: .github/workflows/test-windows.yml [hunk 3]
```diff
@@ -13,7 +13,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        node: [12, 10, 8, 6, 4]
+        node: [15, 14, 12, 10]
```
Comment (jimmywarting) [inline .github/workflows/test-windows.yml]:
```diff
-        node: [15, 14, 12, 10]
+        node: [14, 16, 18]
```

File: index.js [hunk 4]
```diff
@@ -1,21 +1,7 @@
...
-}
-
-// The method startsWith is not defined on string objects in node 0.10
-// eslint-disable-next-line no-extend-native
-String.prototype.startsWith = function(suffix) {
-  return this.substring(0, suffix.length) === suffix;
-};
-
-var pidtree = require('./lib/pidtree');
+const util = require('util');
```
Comment (jimmywarting) [inline index.js]:
```diff
-const util = require('util');
+const util = require('node:util');
```

File: package.json [hunk 5]
```diff
@@ -39,7 +39,7 @@
     "index.d.ts"
   ],
   "engines": {
-    "node": ">=0.10"
+    "node": ">= 10"
```
Comment (jimmywarting) [inline package.json]:
```diff
-    "node": ">= 10"
+    "node": ">= 14"
```

**Review by jimmywarting — COMMENTED** [review]
update versions

**Comment by Kikobeats** [comment]
Applied suggestions.

Can you take care about this PR? It's open for more than one year.

**Comment by jimmywarting** [comment]
> Can you take care about this PR?

Sry, can't... Don't have access to this repo

**Review by jimmywarting — APPROVED** [review]

**Review by huan — CHANGES_REQUESTED** [review]
Thanks for continue improving this PR!

Could you please bump the minor version to the next one (`0.6.0` -> `0.7.0`) so that we can make this change not affect the previous version users?

**Comment by Kikobeats** [comment]
I prefer to don't bump numbers in a PR; That's a thing can be done after merge the PR.

It looks like nobody has the ability to move this forward, so pinging @simonepri

**Comment by MikeMcC399** [comment]
@Kikobeats 

For some reason your PR has remained open without being merged or rejected. In the meantime Node.js versions have moved on.

- I've submitted PR #25 to update the Node.js versions. This would replace some of the changes in this PR. I did not make any changes to `index.js` or `package.json`, so these proposed changes would still be relevant.

**Comment by Kikobeats** [comment]
closing since authors are not interested

**Pull request closed without merging**